var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -6,40 +6,214 @@
+    version("3.0.5", sha256="81ff1b8f10909eeda2d888f97f615e5b92aa26c101a32f36f346f78aa7638ba9")
+    version("3.0.4", sha256="b01714e734b0aa5b0b52f3d7871e4bc9f6c81f25a444b1287e3f2e98a4f5d289")
+    version("3.0.3", sha256="a4ae83cdb2c5029f152064f163d1b66b265980793d76b79908c073027401d2fc")
+    version("3.0.2", sha256="b0c3aad9cac18d0ca73c78c69326031407af6a4bfcc953b3a27506c298bc59a3")
+    version("3.0.1", sha256="812941146c31ab53c9a7695abbca6bfb36ae55878e38f8d0f13a17f3d9c36dc0")
+    version("3.0.0", sha256="1c1299f53788beb2f6b6180d8484d584b6f293c81cea25f45b9632dd608ba4f9")
+    version(
+        "2.3.5",
+        sha256="81c44be76c820b59a8b95f2b398c772160068a5e323f319e33704ff87dbefc5e",
+        deprecated=True,
+    )
+    version(
+        "2.3.4",
+        sha256="16b817311c8055cf055021d895642fbb913585f08ec96f9aff3e0ef20148ec0e",
+        deprecated=True,
+    )
+    version(
+        "2.3.3",
+        sha256="a1246c4a6de385826cae15d9a446e862331eb189a592622b1014c7b49504c908",
+        deprecated=True,
+    )
+    version(
+        "2.3.2",
+        sha256="edcaae88e51360548bbdf1bcbd5048bec0d309b4fe11d8d11d5425afc001e9a1",
+        deprecated=True,
+    )
+    version(
+        "2.3.1",
+        sha256="e9522469f499bcb1c57ec05e9866403e4142c97762d7662b4b638f7493bf5122",
+        deprecated=True,
+    )
+    version(
+        "2.3.0",
+        sha256="c5a803a1a596983681d65aff46762f2f6b82b5f50449a53d0ec7dad368d35842",
+        deprecated=True,
+    )
+    version(
+        "2.2.4",
+        sha256="1f3923d32c392fd2564d137b076134aebdc461dc3148c535a00f9b10a615157a",
+        deprecated=True,
+    )
+    version(
+        "2.2.3",
+        sha256="88dbf6a3bdd907df3e097637ec8e51fde13e2f5e0b44f3667443195481320edf",
+        deprecated=True,
+    )
+    version(
+        "2.2.2",
+        sha256="4a7cf2ec6e43dd5c53d5f877a941367074f4714d93c1977a719782957916169e",
+        deprecated=True,
+    )
+    version(
+        "2.2.1",
+        sha256="14feca1d8d1ce025d3f263b85ebfbebc1a1efff704b6490e95b07603c55c1d63",
+        deprecated=True,
+    )
+    version(
+        "2.2.0",
+        sha256="09d4a2508d903121d29813a85791eeb3a905acbe1c5664b8a88903f8eda64b8f",
+        deprecated=True,
+    )
 
+    variant("mpi", default=True, description="Enable MPI support")
     variant("openmp", default=True, description="Enable OpenMP support")
+    variant(
+        "lcao",
+        default=True,
+        description="Enable Linear Combinition of Atomic Orbital calculation",
+        when="+mpi",
+    )
+    variant("libxc", default=True, description="Support additional functionals via libxc")
+    variant(
+        "elpa",
+        default=True,
+        description="Enable optimised diagonalisation routines from ELPA",
+        when="+lcao",
+    )
+    variant("mathlib", default=False, description="Enable ABACUS's builtin libm")
+    variant(
+        "tests", default=False, description="Build ABACUS unit tests", when="build_system=cmake"
+    )
```

Comment:
Can we avoid the variant and use `self.run_tests` instead? Or is this variant installing something in the prefix?

Reply:
It's a variable that is set to `True` when the package is installed asking for build-time tests. You can grep the builtin repo to see how it's used. One example would be `hdf5`, for instance.

Reply:
According to private discussion with ABACUS developers, unit tests are temporarily removed in this PR.